src/rpc/blockchain.cpp

```diff
@@ -985,6 +987,202 @@ static RPCHelpMan gettxoutsetinfo()
+            filepath = fs::absolute(filepath);
+
+            if (fs::exists(filepath)) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, filepath.u8string() + " already exists. If you are sure this is what you want, move it out of the way first");
+            }
+
+            std::ofstream file;
+            file.open(filepath);
+            if (!file.is_open())
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Cannot open dump file");
+
+            if (!g_coin_stats_index) {
+                throw JSONRPCError(RPC_MISC_ERROR, "Coinstats index is not enabled");
+            }
+
+            const bool cumulative{request.params[1].isNull() ? false : request.params[1].get_bool()};
+
+            file << strprintf("height,"
+                              "block_hash,"
+                              "timestamp,"
+                              "mediantime,"
+                              "version,"
+                              );
+
+            if (cumulative) {
+                file << strprintf("total_txs,"
+                                  "total_txouts,"
+                                  "total_coinbase_amount,"
+                                  "total_subsidy,"
+                                  "total_amount_generated,"
+                                  "total_unclaimed_rewards,"
+                                  "total_amount_spent,"
+                                  "total_new_outputs_ex_coinbase_amount,"
+                                  "total_unspendable_scripts,"
+                                  "total_unspendable_amount,"
+                                  "total_genesis_block,"
+                                  "total_bip30,"
+                                  "total_chainwork,"
+                                  "muhash,"
+                                  "total_bogosize\n");
+            } else {
+                file << strprintf("txs,"
+                                  "txouts,"
+                                  "coinbase_amount,"
+                                  "subsidy,"
+                                  "amount_generated,"
+                                  "unclaimed_rewards,"
+                                  "amount_spent,"
+                                  "new_outputs_ex_coinbase_amount,"
+                                  "unspendable_scripts,"
+                                  "unspendable_amount,"
+                                  "genesis_block,"
+                                  "bip30,"
+                                  "chainwork,"
+                                  "muhash,"
+                                  "bogosize\n");
+            }
+
+            NodeContext& node = EnsureAnyNodeContext(request.context);
+            ChainstateManager& chainman = EnsureChainman(node);
+            CChainState& active_chainstate = chainman.ActiveChainstate();
+            active_chainstate.ForceFlushStateToDisk();
```

Comment:
Pretty sure flushing and getting the cursor will need to happen in the same cs_main scope, otherwise the flush seems best-effort at best?

Reply:
Yeah, I would have described it as best effort so I think it can be omitted here as well. And this made me realize that we don't actually need coins_view and blockman, so I adapted this as well.